crates/vm/levm/src/precompiles.rs

```diff
@@ -1172,11 +1177,119 @@ pub fn bls12_g1add(
+                    ..offset
+                        .checked_add(128)
+                        .ok_or(InternalError::ArithmeticOperationOverflow)?,
+            )
+            .ok_or(InternalError::SlicingError)?;
+
+        if !x.iter().take(16).all(|x| *x == 0) || !y.iter().take(16).all(|x| *x == 0) {
+            return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+        }
+
+        let scalar_bytes = calldata
+            .get(
+                offset
+                    .checked_add(128)
+                    .ok_or(InternalError::ArithmeticOperationOverflow)?
+                    ..offset
+                        .checked_add(LENGTH_PER_PAIR)
+                        .ok_or(InternalError::ArithmeticOperationOverflow)?,
+            )
+            .ok_or(InternalError::SlicingError)?;
+        let scalar_bytes: [u8; 32] = scalar_bytes
+            .try_into()
+            .map_err(|_| PrecompileError::ParsingInputError)?;
+
+        let mut scalar_le = [0u64; 4];
+        for (j, chunk) in scalar_bytes.chunks(8).enumerate() {
+            let bytes: [u8; 8] = chunk
+                .try_into()
+                .map_err(|_| PrecompileError::ParsingInputError)?;
+            if let Some(value) = scalar_le.get_mut(j) {
+                *value = u64::from_be_bytes(bytes);
+            } else {
+                return Err(VMError::Internal(InternalError::SlicingError));
+            }
+        }
+        scalar_le.reverse();
+        let scalar = Scalar::from_raw(scalar_le);
+
+        let x = x.get(16..).ok_or(InternalError::SlicingError)?;
+        let y = y.get(16..).ok_or(InternalError::SlicingError)?;
+        let mut g1_point_byte = Vec::with_capacity(96);
+        g1_point_byte.extend_from_slice(x);
+        g1_point_byte.extend_from_slice(y);
+
+        let g1_point_byte: [u8; 96] = g1_point_byte
+            .try_into()
+            .map_err(|_| PrecompileError::ParsingInputError)?;
+
+        let g1 = if g1_point_byte.iter().all(|e| *e == 0) {
+            G1Projective::identity()
+        } else {
+            let g1 = G1Affine::from_uncompressed(&g1_point_byte);
+            let g1: G1Projective = if g1.is_some().into() {
+                g1.unwrap()
+            } else {
+                return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+            }
+            .into();
+            g1
+        };
+
+        let scaled_point = G1Projective::mul(g1, scalar);
+        result = result.add(&scaled_point);
+    }
+
+    if result.is_identity().into() {
+        let output = [0u8; 128];
+        return Ok(Bytes::copy_from_slice(&output));
+    }
+    let result_bytes = G1Affine::from(result).to_uncompressed();
+
+    let mut output = [0u8; 128];
+    let (x_bytes, y_bytes) = result_bytes
+        .split_at_checked(48)
+        .ok_or(InternalError::SlicingError)?;
+    output[16..64].copy_from_slice(x_bytes);
+    output[80..128].copy_from_slice(y_bytes);
```

Comment:
Nit. You can define `output` once.
```suggestion
    let mut output = [0u8; 128];
    
    if result.is_identity().into() {
        return Ok(Bytes::copy_from_slice(&output));
    }

    let (x_bytes, y_bytes) = G1Affine::from(result)
        .to_uncompressed()
        .split_at_checked(48)
        .ok_or(InternalError::SlicingError)?;

    output[16..64].copy_from_slice(x_bytes);
    output[80..128].copy_from_slice(y_bytes);
```

Reply:
Also, define a constant for the value `48`.

Reply:
Done in [ac65189](https://github.com/lambdaclass/ethrex/pull/1704/commits/ac65189f88a8eaa8ea72704d740867310e18de6b)

Reply:
Here i defined output once as you mentioned [01c0bf4](https://github.com/lambdaclass/ethrex/pull/1704/commits/01c0bf4907405b060d8460387a351f16fd4013f5). I did not commit the suggestion because it was giving clippy lints. But if you think that it must be like the suggestion i will apply those changes.